### PR TITLE
Cms deid 12 caid 16 care (ticket:5456)

### PR DESCRIPTION
**Pull request by contactlp**

### CODE REVIEW

File: deid/bene_id_mapping.sql [hunk 1]
```diff
@@ -37,10 +37,16 @@ from (
   select /*+ PARALLEL(OUTPATIENT_CONDITION_CODES,12) */ distinct bene_id from OUTPATIENT_CONDITION_CODES union
   select /*+ PARALLEL(BCARRIER_DEMO_CODES,12) */ distinct bene_id from BCARRIER_DEMO_CODES union
   select /*+ PARALLEL(HHA_DEMO_CODES,12) */ distinct bene_id from HHA_DEMO_CODES union
-  select /*+ PARALLEL(HOSPICE_DEMO_CODES,12) */ distinct bene_id from HOSPICE_DEMO_CODES
+  select /*+ PARALLEL(HOSPICE_DEMO_CODES,12) */ distinct bene_id from HOSPICE_DEMO_CODES union
+  select /*+ PARALLEL(MAXDATA_IP,12) */ distinct bene_id from MAXDATA_IP union
+  select /*+ PARALLEL(MAXDATA_LT,12) */ distinct bene_id from MAXDATA_LT union
+  select /*+ PARALLEL(MAXDATA_OT,12) */ distinct bene_id from MAXDATA_OT union
+  select /*+ PARALLEL(MAXDATA_PS,12) */ distinct bene_id from MAXDATA_PS union
+  select /*+ PARALLEL(MAXDATA_RX,12) */ distinct bene_id from MAXDATA_RX union
```
Comment (dckc) [inline deid/bene_id_mapping.sql]: Why are MAXDATA_IP ... MAXDATA_RX added here? 3f8c900 says "adde tables for 12caid and 16care " but these are not new; they were in the orginal 2011-2013 data.

File: deid/bene_id_mapping.sql [hunk 2]
```diff
@@ -78,25 +84,149 @@ from (
...
```
Comment (dckc) [inline deid/bene_id_mapping.sql]: Again why are MAXDATA_IP ... MAXDATA_RX added here?

File: deid/cms_deid.sql [hunk 3]
```diff
@@ -203,6 +203,7 @@ select /*+ PARALLEL(outpatient_base_claims_k,12) */
   idt.CLM_NEXT_GNRTN_ACO_IND_CD4, -- Claim Next Generation Accountable Care Organization Indicator Code 4
   idt.CLM_NEXT_GNRTN_ACO_IND_CD5, -- Claim Next Generation Accountable Care Organization Indicator Code 5
   idt.ACO_ID_NUM, -- Claim Accountable Care Organization (ACO) Identification Number
+  idt.CLM_BENE_ID_TYPE_CD,
```
Comment (dckc) [inline deid/cms_deid.sql]: Did you fix this by hand? Is there still a bug in the code that generates the SQL?

File: deid/bene_id_mapping.sql (hunk 1 above)
Comment (dckc) [reply deid/bene_id_mapping.sql]: I see a comment below about "in order for the process to work for the sample data"; is that why?

File: deid/bene_id_mapping.sql [hunk 4]
```diff
@@ -78,25 +84,149 @@ from (
...
+  union
+  select /*+ PARALLEL(MAXDATA_OT,12) */ distinct bene_id, msis_id, state_cd from MAXDATA_OT where bene_id is not null
+  union
+  select /*+ PARALLEL(MAXDATA_RX,12) */ distinct bene_id, msis_id, state_cd from MAXDATA_RX where bene_id is not null
+  ) umid;
+commit;
+create unique index msis_person_mid_st_idx on msis_person (msis_id, state_cd);
+
+
+------------------------------------------------------------------------------------------
```
Comment (dckc) [inline deid/bene_id_mapping.sql]: not cricitcal, but I'm not a fan of ASCII-art comments. And without some text telling me what the line separates, I'm not sure what it's there for.

My preference is JavaDoc-style comment blocks above things to say what they are:

```
/**
 * Bubblegum - a yummy chewey thing.
 */
```

File: deid/cms_deid.sql [hunk 5]
```diff
@@ -1519,4 +1528,1060 @@ select /*+ PARALLEL(hospice_demo_codes,12) */
   idt.EXTRACT_DT
 from hospice_demo_codes idt
 join bene_id_mapping bm on bm.bene_id = idt.bene_id;
-commit;
\ No newline at end of file
+commit;
+
+
+insert /*+ APPEND */ into "&&deid_schema".maxdata_ip
```
Comment (dckc) [inline deid/cms_deid.sql]: `maxdata_ip` isn't new. Why is there all this new code for it? I don't see any explanation in the c2e4747 commit message.

File: deid/cms_mapping_init.sql [hunk 6]
```diff
@@ -17,7 +17,7 @@ create sequence bene_id_deid_seq
 
 -- De-identified msis_id (1:1 msis_id to sequence number mapping)
 create sequence msis_id_deid_seq
-  start with 1
+  start with &&msis_id_deid_seq_start
```
Comment (dckc) [inline deid/cms_mapping_init.sql]: Where is `msis_id_deid_seq_start` defined? The README documents some parameters; should this be documented there?

File: deid/events.sql [hunk 7]
```diff
@@ -593,4 +593,147 @@ select /*+ PARALLEL(bcarrier_demo_codes,12) */
...
+  msis_id,
+  state_cd,
+  EL_DOB,
+  PYMT_DT,
+  ADMSN_DT,
+  SRVC_BGN_DT,
+  SRVC_END_DT,
+  PRNCPL_PRCDR_DT,
+  EXTRACT_DT
+from maxdata_ip
```
Comment (dckc) [inline deid/events.sql]: Again, why all the new code for medicaid tables that have been there from the start?

File: deid/hipaa_dob_shift.sql [hunk 8]
```diff
@@ -42,9 +42,26 @@ select
 from
     min_max_date_events cmm
 left join
-    "&&prev_cms_id_schema".min_max_date_events pmm
+    "&&prev_cms_id_schema"."&&mn_mx_dt_events_prev_yr_cumul" pmm
```
Comment (dckc) [inline deid/hipaa_dob_shift.sql]: Where is `mn_mx_dt_events_prev_yr_cumul` defined / documented?

File: deid/bene_id_mapping.sql (hunk 1 above)
Comment (contactlp) [reply deid/bene_id_mapping.sql]: It is true those tables were there for 2011-13, but not for 2014 and 2015, I believe they got removed from master branch when 2014 code got merged.  To keep these changes more organized, we have used tags/release feature. https://github.com/kumc-bmi/grouse/releases

File: deid/bene_id_mapping.sql (hunk 4 above)
Comment (contactlp) [reply deid/bene_id_mapping.sql]: removed ASCII-art comment.

File: deid/cms_mapping_init.sql (hunk 6 above)
Comment (contactlp) [reply deid/cms_mapping_init.sql]: made inline comment as well as added in README

File: deid/hipaa_dob_shift.sql (hunk 8 above)
Comment (contactlp) [reply deid/hipaa_dob_shift.sql]: Documented the table definition with example.

**Pull request merged**